### DISCUSSION TITLE
Issue: AAPT2 Process Unexpectedly Exit After Merging PR #5604

### DISCUSSION THREAD

**Discussion by TanishMoral11** [oppia/oppia-android#5632] [Q&A (Other)] [answered]
Hello everyone,

After the PR [#5604](https://github.com/oppia/oppia-android/pull/5604) was merged, I started facing an issue in Android Studio when I updated my `develop` branch, as well as other branches. Upon building the project, I encountered the following error:

![image](https://github.com/user-attachments/assets/ae86f208-0bc3-4f5b-9af7-ec24bfee0e36)

`AAPT2 process unexpectedly exit. Error output: aapt2.exe W 12-30 20:33:18 20424 5956 LoadedArsc.cpp:682] Unknown chunk type '200'.`

I have already attempted the following solutions:

- Followed the suggestion in [this StackOverflow answer](https://stackoverflow.com/a/77271439).
- Tried the common solutions like:
  - Cleaning and rebuilding the project.
  - Invalidating the cache and restarting Android Studio.

Unfortunately, none of these approaches have worked, and I am now stuck, unable to make any further progress.

I believe this might be a common issue for anyone who has updated their branch with the latest changes. I would greatly appreciate it if someone could provide guidance or share any suggestions to resolve this problem.

Thank you in advance for your time and assistance!

**Comment by tobioyelekan** [comment]
Currently having the same issue

**Reply by TanishMoral11** [reply]
Hey @tobioyelekan , Have you resolved this issue?

**Reply by tobioyelekan** [reply]
Unfortunately, I have not had the chance to do what Peres recommended, I'll try out too when I get the chance to.

**Comment by adhiamboperes** [comment]
@TanishMoral11, @tobioyelekan, PTAL at  https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions. to setup Bazel, and refer to https://github.com/oppia/oppia-android/discussions/5643 to set up the Bazel plugin in order to be able to continue building the project (and using newwer versions of Android Studio).

**Reply by TanishMoral11** [reply]
Hey @adhiamboperes @Rd4dev , 

**Operating System**:  Windows  
**Android Studio Version**:  LadyBug  
**RAM Size**: 16GB

I have followed all the steps you mentioned, and as you can see from the attached image, Bazel has been successfully installed on my system. Additionally, I’ve installed API levels 34, 33, 32, 30, 29, and 28 (just to be on the safe side).
![Image](https://github.com/user-attachments/assets/6435a2e0-db90-4f09-97b2-9af93a21de4b)

![Image](https://github.com/user-attachments/assets/bd175e6b-6f51-4f3c-8c38-d83ad74b34ce)


However, when I run the following command using the Ubuntu terminal within Android Studio.:  
`bazel build //:oppia`,  
I am encountering the following error:

![Image](https://github.com/user-attachments/assets/43717335-d731-456d-a897-2fde1e156635)

```bash 
ERROR: /home/tanishmoral/.cache/bazel/_bazel_tanishmoral/2014fc05ae87ff7c07732d3eb414456a/external/bazel_tools/tools/android/BUILD:234:8: @bazel_tools//tool
s/android:desugar_java8_legacy_libs depends on @android_tools//:desugar_jdk_libs.jar in repository @android_tools which failed to fetch. no such package '@a
ndroid_tools//': The repository's path is "android_tools" (absolute: "/home/tanishmoral/opensource/oppia-bazel-tools") but it does not exist or is not a directory.
ERROR: Analysis of target '//:oppia' failed; build aborted:                                                                                                 
INFO: Elapsed time: 1.563s                                                                                                                                  
INFO: 0 processes.                                                                                                                                          
FAILED: Build did NOT complete successfully (0 packages loaded, 5 targets configured)                                                                       
    currently loading: app ... (2 packages)                                                                                                                 
    Fetching repository @remote_java_tools; starting     
  ```  
    
    
Could you kindly guide me on the next steps to resolve this issue? Any help would be greatly appreciated.
Thanks !!

**Reply by Rd4dev** [reply]
Hi @TanishMoral11, since the log mentioned 
```
(absolute: "/home/tanishmoral/opensource/oppia-bazel-tools") but it does not exist or is not a directory
```
would it be possible to confirm if you have the `oppia-android-tools` directory setup in the `opensource/` folder. This is created using the `setup.sh` script. (Ref: https://github.com/oppia/oppia-android/wiki/Bazel-Setup-Instructions-for-Windows#5-preparing-build-environment-for-oppia-android)


![image](https://github.com/user-attachments/assets/ae5b7056-4101-4e57-8d67-d274ce2fbb62)

**Reply by Rd4dev** [reply]
Also, running `bazel clean --expunge` may help clear out any stale configurations and facilitate a clean rebuild.

**Reply by TanishMoral11** [reply]
Hi @Rd4dev ,

Thank you for your suggestion. As per your guidance, I have confirmed that the `oppia-android-tools` directory is indeed present in my `opensource` folder.

![Image](https://github.com/user-attachments/assets/08562f41-aacb-499a-b395-3ac293489cc6)

 I’ve followed the setup instructions provided in the [Bazel Setup Instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions) and the [Bazel Setup Instructions for Windows](https://github.com/oppia/oppia-android/wiki/Bazel-Setup-Instructions-for-Windows).

However, despite completing all the steps, I am still unsure why the setup isn’t working properly. The issue persists, 


![Image](https://github.com/user-attachments/assets/016b5b9c-1b1e-43b4-afa5-6e65214d9f75)

```
Starting local Bazel server and connecting to it...
ERROR: /home/tanishmoral/.cache/bazel/_bazel_tanishmoral/57a402bbe3009b818831969a3a323e61/external/bazel_tools/tools/android/BUILD:234:8: @bazel_tools//tools/android:desugar_java8_legacy_libs depends on @android_tools//:desugar_jdk_libs.jar in repository @android_tools which failed to fetch. no such package '@android_tools//': The repository's path is "android_tools" (absolute: "/home/tanishmoral/opensource/oppia-bazel-tools") but it does not exist or is not a directory.
ERROR: Analysis of target '//:oppia' failed; build aborted:
INFO: Elapsed time: 17.093s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (55 packages loaded, 234 targets configured)
    currently loading: scripts ... (2 packages)

```

and I am unable to make any further progress due to this Bazel-related error. Additionally, I’m encountering the following error message when I try to update my code:

```
AAPT2 process unexpectedly exit. Error output: aapt2.exe W 12-30 20:33:18 20424 5956 LoadedArsc.cpp:682] Unknown chunk type '200'.
```

@adhiamboperes, Because of this issue, I have been unable to merge my code with the latest changes from the `develop` branch.

Could you please assist me further in resolving this issue? Any additional steps or guidance you can provide to help resolve this Issue .

Thanks !!

**Reply by Rd4dev** [reply]
Today I tried to do a fresh re-build of the project and was able to build it with the wsl. Can you please try with a fresh setup just to make sure none of the old configurations are affecting the build.

These are the steps I followed:

From cmd I Installed ubuntu 22.04
```sh
wsl --install Ubuntu-22.04

set username 
password
```

Open the ubuntu 22.04 terminal

Bazel instruction with wsl Ubuntu-22.04

```sh
$ sudo apt update && sudo apt upgrade

$ sudo apt install openjdk-17-jdk
$ java --version
openjdk 17.0.10 2024-01-16
OpenJDK Runtime Environment (build 17.0.10+7-Ubuntu-122.04.1)
OpenJDK 64-Bit Server VM (build 17.0.10+7-Ubuntu-122.04.1, mixed mode, sharing)

$ sudo apt install python2
$ sudo apt install gcc

$ mkdir -p $HOME/Android/Sdk
```

Download the command line tools from https://developer.android.com/studio#command-tools

```sh
$ cp /mnt/c/Users/<Name>/Downloads/commandlinetools*.zip $HOME/Android/Sdk

$ cd $HOME/Android/Sdk
$ sudo apt install unzip
$ unzip commandlinetools*.zip

$ cd cmdline-tools/
$ mkdir tools
$ mv -i * tools
$ cd ..

$ echo "export ANDROID_HOME=\$HOME/Android/Sdk" >> ~/.bashrc
$ echo "export PATH=\$ANDROID_HOME/cmdline-tools/tools/bin/:\$PATH" >> ~/.bashrc
$ source ~/.bashrc

$ sdkmanager
$ sdkmanager --install "platform-tools"
$ sdkmanager --install "platforms;android-34" [updated version]
$ sdkmanager --install "build-tools;32.0.0"

$ sudo apt install apt-transport-https curl gnupg -y
$ curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor >bazel-archive-keyring.gpg
$ sudo mv bazel-archive-keyring.gpg /usr/share/keyrings
$ echo "deb [arch=amd64 signed-by=/usr/share/keyrings/bazel-archive-keyring.gpg] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list

$ sudo apt update && sudo apt install bazel
$ bazel --version
bazel 8.0.0 [will default to install the latest version]

$ sudo apt install bazel-6.5.0
$ bazel --version
bazel 8.0.0

$ sudo rm /usr/bin/bazel
$ sudo ln -s /usr/bin/bazel-6.5.0 /usr/bin/bazel
$ bazel --version
bazel 6.5.0

$ mkdir $HOME/opensource
$ cd $HOME/opensource

$ git clone https://github.com/<GitHub-Username>/oppia-android.git

$ cd oppia-android/
$ scripts/setup.sh
-bash: scripts/setup.sh: Permission denied

$ chmod +x scripts/setup.sh
$ scripts/setup.sh
Using Ktlint 0.37.1
Ktlint file downloaded
Using Checkstyle version 8.37
Checkstyle file downloaded
Using Bufbuild version v0.37.1
Buf downloaded
Using Buildifier version 3.4.0
Buildifier file downloaded

$ bazel build //:oppia
Target //:oppia up-to-date:
  bazel-bin/oppia_deploy.jar
  bazel-bin/oppia_unsigned.apk
  bazel-bin/oppia.apk
```

<details>
<summary>My setup</summary>

```sh

rddev@DESKTOP-CLN30QH:~/opensource$ python --version
Python 2.7.18

rddev@DESKTOP-CLN30QH:~/opensource$ java --version
openjdk 17.0.13 2024-10-15
OpenJDK Runtime Environment (build 17.0.13+11-Ubuntu-2ubuntu122.04)
OpenJDK 64-Bit Server VM (build 17.0.13+11-Ubuntu-2ubuntu122.04, mixed mode, sharing)

rddev@DESKTOP-CLN30QH:~/opensource$ bazel --version
bazel 6.5.0

rddev@DESKTOP-CLN30QH:~/opensource$ sdkmanager --list
Warning: Errors during XML parse:
Warning: Additionally, the fallback loader failed to parse the XML.
Installed packages:=====================] 100% Computing updates...
  Path                                     | Version | Description                             | Location
  -------                                  | ------- | -------                                 | -------
  build-tools;29.0.2                       | 29.0.2  | Android SDK Build-Tools 29.0.2          | build-tools/29.0.2/
  build-tools;32.0.0                       | 32.0.0  | Android SDK Build-Tools 32              | build-tools/32.0.0/
  emulator                                 | 34.2.14 | Android Emulator                        | emulator/
  platform-tools                           | 35.0.1  | Android SDK Platform-Tools              | platform-tools/
  platforms;android-33                     | 3       | Android SDK Platform 33                 | platforms/android-33/
  platforms;android-34                     | 3       | Android SDK Platform 34                 | platforms/android-34/
  system-images;android-23;default;x86     | 10      | Intel x86 Atom System Image             | system-images/android-23/default/x86/
  system-images;android-23;google_apis;x86 | 33      | Google APIs Intel x86 Atom System Image | system-images/android-23/google_apis/x86/
```

File tree

```sh
rddev@DESKTOP-CLN30QH:~/opensource$ ls $HOME/Android/Sdk
build-tools  cmdline-tools  emulator  licenses  platform-tools  platforms  system-images  tools

rddev@DESKTOP-CLN30QH:~/opensource$ ls
oppia-android  oppia-android-tools

rddev@DESKTOP-CLN30QH:~/opensource$ ls oppia-android
BUILD.bazel  app                  bazel-testlogs     build_vars.bzl         domain             gradlew.bat                    oppia_android_test.bzl  third_party  wiki
LICENSE      bazel-bin            buf.yaml           bundle_config.pb.json  gradle             instrumentation                scripts                 tools
NOTICE       bazel-oppia-android  build.gradle       config                 gradle.properties  model                          settings.gradle         utility
WORKSPACE    bazel-out            build_flavors.bzl  data                   gradlew            oppia_android_application.bzl  testing                 version.bzl
```

</details>

And the generated `oppia.apk` should be a fully functioning development version of the app that can be installed using `adb` - though I had to copy it over to the windows path to install from the Windows cmd as wsl wasn't able to detect the connected devices.

Can you confirm if the issue persists with a fresh rebuild too?

**Reply by Rd4dev** [reply]
And regarding: `AAPT2 process unexpectedly exit. Error output: aapt2.exe W 12-30 20:33:18 20424 5956 LoadedArsc.cpp:682] Unknown chunk type '200'.` — this issue will still persist, and we will need to install the Bazel plugin to import it as Bazel project in Android Studio to properly run the app.

However, I haven’t been able to get it fully set up with Android Studio yet due to file path issues on Windows. I’ll post updates if I manage to find a solution for this.

> If anyone has found a solution, that would be great. Otherwise, for a quick band-aid fix, one can build the APK as described above and deploy it using the `adb` tool (though it’s not the most efficient method).

**Reply by TanishMoral11** [reply]
> Today I tried to do a fresh re-build of the project and was able to build it with the wsl. Can you please try with a fresh setup just to make sure none of the old configurations are affecting the build.
> 
> These are the steps I followed:
> 
> From cmd I Installed ubuntu 22.04
> 
> ```shell
> wsl --install Ubuntu-22.04
> 
> set username 
> password
> ```
> 
> Open the ubuntu 22.04 terminal
> 
> Bazel instruction with wsl Ubuntu-22.04
> 
> ```shell
> $ sudo apt update && sudo apt upgrade
> 
> $ sudo apt install openjdk-17-jdk
> $ java --version
> openjdk 17.0.10 2024-01-16
> OpenJDK Runtime Environment (build 17.0.10+7-Ubuntu-122.04.1)
> OpenJDK 64-Bit Server VM (build 17.0.10+7-Ubuntu-122.04.1, mixed mode, sharing)
> 
> $ sudo apt install python2
> $ sudo apt install gcc
> 
> $ mkdir -p $HOME/Android/Sdk
> ```
> 
> Download the command line tools from https://developer.android.com/studio#command-tools
> 
> ```shell
> $ cp /mnt/c/Users/<Name>/Downloads/commandlinetools*.zip $HOME/Android/Sdk
> 
> $ cd $HOME/Android/Sdk
> $ sudo apt install unzip
> $ unzip commandlinetools*.zip
> 
> $ cd cmdline-tools/
> $ mkdir tools
> $ mv -i * tools
> $ cd ..
> 
> $ echo "export ANDROID_HOME=\$HOME/Android/Sdk" >> ~/.bashrc
> $ echo "export PATH=\$ANDROID_HOME/cmdline-tools/tools/bin/:\$PATH" >> ~/.bashrc
> $ source ~/.bashrc
> 
> $ sdkmanager
> $ sdkmanager --install "platform-tools"
> $ sdkmanager --install "platforms;android-34" [updated version]
> $ sdkmanager --install "build-tools;32.0.0"
> 
> $ sudo apt install apt-transport-https curl gnupg -y
> $ curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor >bazel-archive-keyring.gpg
> $ sudo mv bazel-archive-keyring.gpg /usr/share/keyrings
> $ echo "deb [arch=amd64 signed-by=/usr/share/keyrings/bazel-archive-keyring.gpg] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
> 
> $ sudo apt update && sudo apt install bazel
> $ bazel --version
> bazel 8.0.0 [will default to install the latest version]
> 
> $ sudo apt install bazel-6.5.0
> $ bazel --version
> bazel 8.0.0
> 
> $ sudo rm /usr/bin/bazel
> $ sudo ln -s /usr/bin/bazel-6.5.0 /usr/bin/bazel
> $ bazel --version
> bazel 6.5.0
> 
> $ mkdir $HOME/opensource
> $ cd $HOME/opensource
> 
> $ git clone https://github.com/<GitHub-Username>/oppia-android.git
> 
> $ cd oppia-android/
> $ scripts/setup.sh
> -bash: scripts/setup.sh: Permission denied
> 
> $ chmod +x scripts/setup.sh
> $ scripts/setup.sh
> Using Ktlint 0.37.1
> Ktlint file downloaded
> Using Checkstyle version 8.37
> Checkstyle file downloaded
> Using Bufbuild version v0.37.1
> Buf downloaded
> Using Buildifier version 3.4.0
> Buildifier file downloaded
> 
> $ bazel build //:oppia
> Target //:oppia up-to-date:
>   bazel-bin/oppia_deploy.jar
>   bazel-bin/oppia_unsigned.apk
>   bazel-bin/oppia.apk
> ```
> 
> My setup
> And the generated `oppia.apk` should be a fully functioning development version of the app that can be installed using `adb` - though I had to copy it over to the windows path to install from the Windows cmd as wsl wasn't able to detect the connected devices.
> 
> Can you confirm if the issue persists with a fresh rebuild too?

Hey @Rd4dev , 
Thank you so much for your help! I decided to try a fresh installation again, and thankfully, the previous error is gone (hopefully some progress has been made :> ). 
However, I’m now facing the following issue:


``` sh
tanishmoral@Tanish:~/opensource/oppia-android$ bazel build //:oppia && adb install -r bazel-bin/oppia.apk
INFO: Analyzed target //:oppia (361 packages loaded, 6925 targets configured).
INFO: Found 1 target...
ERROR: /home/tanishmoral/.cache/bazel/_bazel_tanishmoral/57a402bbe3009b818831969a3a323e61/external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/BUILD:3:13: Building external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/libr8.jar (16 source files) and running annotation processors (AutoValueProcessor) [for tool] failed: (Exit 1): java failed: error executing command (from target @bazel_tools//src/tools/android/java/com/google/devtools/build/android/r8:r8) external/remotejdk11_linux/bin/java -XX:-CompactStrings '--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED' ... (remaining 18 arguments skipped)
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CompatDexBuilder.java:38: error: cannot find symbol
import com.google.devtools.build.lib.worker.ProtoWorkerMessageProcessor;
                                           ^
  symbol:   class ProtoWorkerMessageProcessor
  location: package com.google.devtools.build.lib.worker
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CompatDexBuilder.java:39: error: cannot find symbol
import com.google.devtools.build.lib.worker.WorkRequestHandler;
                                           ^
  symbol:   class WorkRequestHandler
  location: package com.google.devtools.build.lib.worker
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:36: error: cannot find symbol
import com.google.devtools.build.lib.worker.ProtoWorkerMessageProcessor;
                                           ^
  symbol:   class ProtoWorkerMessageProcessor
  location: package com.google.devtools.build.lib.worker
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:37: error: cannot find symbol
import com.google.devtools.build.lib.worker.WorkRequestHandler;
                                           ^
  symbol:   class WorkRequestHandler
  location: package com.google.devtools.build.lib.worker
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/OptionsConverters.java:27: error: cannot find symbol
  public static class PathConverter extends Converter.Contextless<Path> {
                                                     ^
  symbol:   class Contextless
  location: interface Converter
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CoreLibraryDesugar.java:58: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CoreLibraryDesugar.java:70: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CoreLibraryDesugar.java:83: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CoreLibraryDesugar.java:96: error: incompatible types: Class<PathConverter> cannot be converted to Class<? extends Converter>
        converter = PathConverter.class,
                                 ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CoreLibraryDesugar.java:124: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:75: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:89: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:102: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:143: error: incompatible types: Class<PathConverter> cannot be converted to Class<? extends Converter>
        converter = PathConverter.class,
                                 ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:364: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/DexFileMerger.java:116: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/DexFileMerger.java:129: error: incompatible types: Class<PathConverter> cannot be converted to Class<? extends Converter>
        converter = PathConverter.class,
                                 ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/DexFileMerger.java:150: error: incompatible types: Class<ExistingPathConverter> cannot be converted to Class<? extends Converter>
        converter = ExistingPathConverter.class,
                                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CompatDexBuilder.java:170: error: cannot find symbol
        WorkRequestHandler workerHandler =
        ^
  symbol:   class WorkRequestHandler
  location: class CompatDexBuilder
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CompatDexBuilder.java:171: error: package WorkRequestHandler does not exist
            new WorkRequestHandler.WorkRequestHandlerBuilder(
                                  ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CompatDexBuilder.java:172: error: package WorkRequestHandler does not exist
                    new WorkRequestHandler.WorkRequestCallback(
                                          ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CompatDexBuilder.java:177: error: cannot find symbol
                    new ProtoWorkerMessageProcessor(System.in, realStdOut))
                        ^
  symbol:   class ProtoWorkerMessageProcessor
  location: class CompatDexBuilder
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/CompatDexBuilder.java:394: error: cannot find symbol
      return new AutoValue_CompatDexBuilder_DexingKeyR8(
                 ^
  symbol:   class AutoValue_CompatDexBuilder_DexingKeyR8
  location: class DexingKeyR8
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/desugar/OutputConsumer.java:209: error: cannot find symbol
        super(Opcodes.ASM9, null);
                     ^
  symbol:   variable ASM9
  location: interface Opcodes
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/OptionsConverters.java:39: error: method does not override or implement a method from a supertype   
    @Override
    ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/OptionsConverters.java:54: error: method does not override or implement a method from a supertype   
    @Override
    ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:685: error: cannot find symbol
      WorkRequestHandler workerHandler =
      ^
  symbol:   class WorkRequestHandler
  location: class Desugar
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:686: error: package WorkRequestHandler does not exist
          new WorkRequestHandler.WorkRequestHandlerBuilder(
                                ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:687: error: package WorkRequestHandler does not exist
                  new WorkRequestHandler.WorkRequestCallback(
                                        ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java:690: error: cannot find symbol
                  new ProtoWorkerMessageProcessor(System.in, realStdOut))
                      ^
  symbol:   class ProtoWorkerMessageProcessor
  location: class Desugar
Target //:oppia failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 29.203s, Critical Path: 9.28s
INFO: 992 processes: 358 internal, 623 linux-sandbox, 11 worker.
FAILED: Build did NOT complete successfully

```

I believe this error might be related to the `WORKSPACE` configuration. Could you please suggest any solutions or areas I should check to resolve this?


Thanks !!

**Reply by TanishMoral11** [reply]
Hey @Rd4dev , 

Thank you for your guidance! I’ve tried all the solutions, and I’ve finally managed to set up **Oppia** in my dual-boot setup. Currently, I am using **Ubuntu 22.04.5**.

---

#### Progress so far:
On Ubuntu, I successfully set up **Bazel** with **Oppia**. However, the issue regarding the `AAPT2 process` still persists. Here’s the error output:  

```text
AAPT2 process unexpectedly exit. Error output: 
aapt2.exe W 12-30 20:33:18 20424 5956 LoadedArsc.cpp:682] Unknown chunk type '200'.
```

As per this [Discussion](https://github.com/oppia/oppia-android/discussions/5643#discussioncomment-11874290), I ran the commands via the terminal, and the app works correctly.

@manas-yu , I Have Some Questions :


1. I downloaded and built the app using:

```bash
bazel build //:oppia && adb install -r bazel-bin/oppia.apk
```

If I switch to another branch (e.g., Branch 2), do I need to run the same command again and then execute the app using:

```bash
adb shell am start -n org.oppia.android/.app.splash.SplashActivity
```
Can you confirm if the following is the correct approach to run the app via terminal, or am I doing something wrong?

```bash
bazel build //:oppia && adb install -r bazel-bin/oppia.apk 
adb shell am start -n org.oppia.android/.app.splash.SplashActivity
```

#### Additional Observations:
I switched to Ubuntu due to Windows’ path-related issues. However, when I run:

```bash
bazel build //:oppia && adb install -r bazel-bin/oppia.apk
```

my laptop freezes, and I am unable to move the cursor or perform any actions.

My system specifications are quite good, so I’m unsure why this happens. Are you facing the same issue?

![Image](https://github.com/user-attachments/assets/7bccf428-385d-4c7c-9305-85301d64da2e)

Thank !!

**Reply by manas-yu** [reply]
Hi @TanishMoral11,
1. Yes, the build command `bazel build //:oppia` needs to be run whenever there are code changes, including when switching branches. It updates the APK at bazel-bin/oppia.apk, which you can then install using `adb install -r`
2. **The first build** with Bazel is resource-intensive and may cause your system to freeze, especially if other processes are running. I recommend closing unnecessary applications and running the build command. Subsequent builds will be faster as Bazel caches intermediate results.

**Reply by Rd4dev** [reply]
Apologies, @TanishMoral11, I couldn’t get to this sooner. Super glad the dual-boot option worked! — it seems like the quickest fix. I’ll look into the Windows fixes and the issue above as soon as I can.

**Reply by adhiamboperes** [reply]
@TanishMoral11, from yesterday's CLaM meeting, we will be building and installing with `bazel run //:install_oppia_dev` going forward.

**Comment by uphargaur** [comment]
Hi @adhiamboperes ,

I am also facing this issue in the development branch only. I tried setting up Bazel as well, but it overloads the RAM, causing my system to hang and eventually quit. Can we consider an alternative for this?

Here are my system specifications:


![image](https://github.com/user-attachments/assets/4f824203-fae7-4392-aec8-ba440f776d42)

**Reply by adhiamboperes** [reply]
@uphargaur, how much RAM and free storage do you have ?

**Comment by BenHenning** [comment, marked as answer]
This issue was caused by an incompatibility with Gradle (it's actually a problem within the AGP toolchain itself). See https://github.com/oppia/oppia-android/pull/5604#issuecomment-2563285961 and surrounding comments for a _lot_ more discussion on the topic.

The issue should now be resolved fully since Gradle has been disabled in the codebase (and will soon be completely removed) as Bazel won't run into this issue.
